src/dmd/dmodule.d

```diff
@@ -535,6 +501,77 @@ extern (C++) final class Module : Package
         return m;
     }
 
+    /**
+     * Converts a chain of identifiers to the filename of the module
+     *
+     * Params:
+     *  packages = the names of the "parent" packages
+     *  ident = the name of the child package or module
+     *
+     * Returns:
+     *  the filename of the child package or module
+     */
+    static const(char)* getFilename(Identifiers* packages, Identifier ident)
+    {
+        const(char)[] filename = ident.toString();
+
+        if (packages == null || packages.dim == 0)
+            return filename.ptr;
+
+        OutBuffer buf;
+        OutBuffer dotmods;
+        auto ms = global.params.modFileAliasStrings;
+        const msdim = ms ? ms.dim : 0;
+
+        void checkModFileAlias(const(char)[] p)
+        {
+            /* Check and replace the contents of buf[] with
+            * an alias string from global.params.modFileAliasStrings[]
```

Comment:
Took me a while to realize this was talking about `ms[]`, not `global.params.modFileAliasStrings[]`